justfile

```
@@ -22,3 +22,13 @@ update-template-lockfile:
     sed -i 's/$proj_name/\{\{ project_name \}\}/' $cargo_lock_path
     cp $cargo_lock_path cot-cli/src/project_template/Cargo.lock.template
     rm -rf $tmpdir
+
+coverage:
+    # generate coverage report as HTML
+    # requires cargo-llvm-cov installed and nightly toolchain
+    cargo llvm-cov --all-features --workspace --branch --doctests --html --open
+
+docs:
+    # generate docs for the `cot` crate with similar settings to docs.rs
+    # requires nightly toolchain
+    RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --no-deps --all-features -p cot
```

I think it would be more useful if it generated docs for all crates, not only `cot`
```suggestion
    RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --no-deps --all-features
```

---

Sadly when you don't specify this, the docs for `cot-cli` binary (which is called `cot`) will override the docs for the `cot` crate, so this makes the whole thing useless. I thought that since we only really care about `cot` docs, it's fine just to generate docs for this single crate.

---

Wouldn't `--lib` flag solve that problem?

---

Fair, this works.